airflow/decorators/__init__.py

```diff
@@ -55,5 +57,249 @@ def python(python_callable: Optional[Callable] = None, multiple_outputs: Optiona
+            The path is also made available via the environment variable
+            ``AIRFLOW_TMP_DIR`` inside the container.
+        :type tmp_dir: str
+        :param user: Default user inside the docker container.
+        :type user: int or str
+        :param volumes: List of volumes to mount into the container, e.g.
+            ``['/host/path:/container/path', '/host/path2:/container/path2:ro']``.
+        :type volumes: list
+        :param working_dir: Working directory to
+            set on the container (equivalent to the -w switch the docker client)
+        :type working_dir: str
+        :param xcom_all: Push all the stdout or just the last line.
+            The default is False (last line).
+        :type xcom_all: bool
+        :param docker_conn_id: ID of the Airflow connection to use
+        :type docker_conn_id: str
+        :param dns: Docker custom DNS servers
+        :type dns: list[str]
+        :param dns_search: Docker custom DNS search domain
+        :type dns_search: list[str]
+        :param auto_remove: Auto-removal of the container on daemon side when the
+            container's process exits.
+            The default is False.
+        :type auto_remove: bool
+        :param shm_size: Size of ``/dev/shm`` in bytes. The size must be
+            greater than 0. If omitted uses system default.
+        :type shm_size: int
+        :param tty: Allocate pseudo-TTY to the container
+            This needs to be set see logs of the Docker container.
+        :type tty: bool
+        :param privileged: Give extended privileges to this container.
+        :type privileged: bool
+        :param cap_add: Include container capabilities
+        :type cap_add: list[str]
+        """
+        return _docker_task(
+            python_callable=python_callable,
+            multiple_outputs=multiple_outputs,
+            image=image,
+            api_version=api_version,
+            container_name=container_name,
+            cpus=cpus,
+            docker_url=docker_url,
+            environment=environment,
+            private_environment=private_environment,
+            force_pull=force_pull,
+            mem_limit=mem_limit,
+            host_tmp_dir=host_tmp_dir,
+            network_mode=network_mode,
+            tls_ca_cert=tls_ca_cert,
+            tls_client_cert=tls_client_cert,
+            tls_client_key=tls_client_key,
+            tls_hostname=tls_hostname,
+            tls_ssl_version=tls_ssl_version,
+            tmp_dir=tmp_dir,
+            user=user,
+            volumes=volumes,
+            working_dir=working_dir,
+            xcom_all=xcom_all,
+            docker_conn_id=docker_conn_id,
+            dns=dns,
+            dns_search=dns_search,
+            auto_remove=auto_remove,
+            shm_size=shm_size,
+            tty=tty,
+            privileged=privileged,
+            cap_add=cap_add,
+            extra_hosts=extra_hosts,
+            **kwargs,
+        )
```

Comment:
If I correctly understand most of those arguments are for DockerOperator. I'm wondering if it would make sense to generate the method automatically using DockerOperator class? This would have to advantages:
- we would be sure that any change in signature in DockerOperator is automatically reflected in `task.docker`
- the mechanism can be reused for other operators/decorators

Of course "explicit is better than implicit" but "missing" arguments that users would like to use are a common case in operators world. WDYT @dimberman ?